dali/test/python/test_operator_roi_random_crop.py

```diff
@@ -0,0 +1,98 @@
+import numpy as np
+import nvidia.dali as dali
+import nvidia.dali.fn as fn
+import nvidia.dali.types as types
+import nvidia.dali.math as math
+from test_utils import check_batch, dali_type
+import random
+from nose.tools import assert_raises
+
+np.random.seed(4321)
+
+def random_shape(min_sh, max_sh, ndim):
+  return np.array(
+      [np.random.randint(min_sh, max_sh) for s in range(ndim)],
+      dtype=np.int32
+    )
+
+def check_roi_random_crop(ndim=2, batch_size=3,
+                          roi_min_start = 0, roi_max_start = 100,
+                          roi_min_extent = 20, roi_max_extent = 50,
+                          crop_min_extent = 20, crop_max_extent = 50,
+                          in_shape_min = 400, in_shape_max = 500,
+                          use_in_shape_arg = False, use_shape_like_in = False,
+                          niter=3):
+    pipe = dali.pipeline.Pipeline(batch_size=batch_size, num_threads=4, device_id=0, seed=1234)
+    with pipe:
+        in_shape_out = None
+        in_shape_arg = None
+        inputs = []
+        if use_in_shape_arg or use_shape_like_in:
+            assert in_shape_min < in_shape_max
+            shape_gen_f = lambda: random_shape(in_shape_min, in_shape_max, ndim)
+            if use_shape_like_in:
+                shape_like_in = dali.fn.external_source(lambda: np.zeros(shape_gen_f()), 
+                                                        device='cpu', batch=False)
+                in_shape_out = dali.fn.shapes(shape_like_in)
+                inputs += [shape_like_in]
+            elif use_in_shape_arg:
+                in_shape_arg = dali.fn.external_source(shape_gen_f, batch=False)
+                in_shape_out = in_shape_arg
+        crop_shape = fn.random.uniform(range=(crop_min_extent, crop_max_extent + 1), 
+                                       shape=(ndim,), dtype=types.INT32, device='cpu')
+        roi_shape = fn.random.uniform(range=(roi_min_extent, roi_max_extent + 1),
+                                      shape=(ndim,), dtype=types.INT32, device='cpu')
+        roi_start = fn.random.uniform(range=(roi_min_start, roi_max_start + 1),
+                                      shape=(ndim,), dtype=types.INT32, device='cpu')
+        crop_start = fn.roi_random_crop(*inputs, crop_shape=crop_shape,
+                                        roi_start=roi_start, roi_shape=roi_shape,
```

Comment:
Can you add a negative cases when both `use_in_shape_arg` and `use_shape_like_in` are provided, as well as `roi_end` with `roi_shape`?